generator/src/main/java/com/scottlogic/deg/generator/utils/RepeatableIterator.java

```diff
@@ -0,0 +1,39 @@
+package com.scottlogic.deg.generator.utils;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+public class RepeatableIterator<T> implements Iterator<T> {
```

Comment:
Naming point.

Maybe to further clarify what this type does, we could call it a `RestartableIterator` then rename the `reset` method to `restart`?
Would this help explain what it is rather than one that repeats its values?

Reply:
To be honest I'm not a massive fan of either names but RestartableIterator does express more clearly what it does/how to use it as opposed to Repeatable which makes it sound like it would be automatic.

Reply:
Also, while I'm making a change to this class. Since DI is incoming soon do we want to introduce a RestartableIterator interface?